fix(curriculum): correct validation example in encapsulation lesson

Jeevankumar-s opened the pull request:
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

Closes #64905

This PR fixes an incorrect example in the Python OOP encapsulation lesson.

- Corrected the deposit example to reflect actual validation behavior
- Ensured ValueError is only raised for negative amounts
- Updated outputs to match real execution
- Content-only change, no logic modifications

Commit:
fix(python): correct validation example in encapsulation lesson

gikf commented:
Hi there,

Thanks for creating this pull request.

The linked issue has not been triaged yet, and a solution has not been agreed upon. Once the issue is open for contribution, you are welcome to update this pull request to reflect the issue consensus. Until the issue is open for contribution, we will not be able to review your pull request.

Dario-DC commented on curriculum/challenges/english/blocks/lecture-understanding-object-oriented-programming-and-encapsulation/68420be9af9d89620af7944a.md:
```diff
@@ -144,11 +144,11 @@ class Wallet:
 
 acct_one.deposit(50)
-print(acct_one.get_balance()) # 50
+print(acct_one.get_balance()) # 53
```
Let's also use `deposit` with a negative argument since the text mentions that it calls `__validate`.
```suggestion
print(acct_one.get_balance()) # 53
acct_one.deposit(-4) # ValueError: Amount must be positive
```

Dario-DC commented on curriculum/challenges/english/blocks/lecture-understanding-object-oriented-programming-and-encapsulation/68420be9af9d89620af7944a.md:
```diff
@@ -144,11 +144,11 @@ class Wallet:
-print(acct_one.get_balance()) # 50
+print(acct_one.get_balance()) # 53
 acct_one.withdraw(-8) # ValueError('Amount must be positive')
 acct_one.withdraw(58) # ValueError('Insufficient funds')
```
Not related with this issue but please update these as well
```suggestion
acct_one.withdraw(-8) # ValueError: Amount must be positive
acct_one.withdraw(58) # ValueError: Insufficient funds
```

Dario-DC reviewed (commented):
Hey @Jeevankumar-s I left a couple of comments. Once those are addressed I'm happy to approve. For the future, please just work on issues that are open for contribution, which have the `help wanted` label.

Jeevankumar-s commented:
> Hey @Jeevankumar-s I left a couple of comments. Once those are addressed I'm happy to approve. For the future, please just work on issues that are open for contribution, which have the `help wanted` label.

Thanks for the review! I’ve addressed all the comments and updated the examples accordingly. I’ll make sure to work only on issues marked open for contribution / help wanted going forward.

Commit:
fix(curriculum): correct validation examples in encapsulation lesson

Commit:
fix(curriculum): correct validation examples in encapsulation lesson

Commit:
fix(curriculum): reorder validation examples to keep errors at the end

majestic-owl448 reviewed (approved):
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉
